File: 백준/Silver/1764. 듣보잡/듣보잡.py
a, b = map(int, input().split())
arr1 = set()
arr2 = set()
ans = []
for i in range(a):
    arr1.add(input())
for i in range(b):
    arr2.add(input())
for i in arr1:
    if i in arr2:
        ans.append(i)
print(len(ans))
ans.sort()
for i in ans:
    print(i)

File: 백준/Silver/1764. 듣보잡/듣보잡.js
const stdin =
  process.platform === "linux"
    ? require("fs").readFileSync("/dev/stdin")
    : `3 4
ohhenrie
charlie
baesangwook
obama
baesangwook
ohhenrie
clinton`;

const input = stdin.toString().trim().split("\n");
const [N, M] = input.shift().split(" ").map(Number);

const first = new Set();
const second = new Set();

let answer = [];

for (let i = 0; i < input.length; i++) {
  if (i < N) {
    first.add(input[i]);
  } else {
    second.add(input[i]);
  }
}

second.forEach((v) => {
  if (first.has(v)) answer.push(v);
});

answer.sort();
console.log(answer.length + '\n' + answer.join('\n'));
